### DISCUSSION TITLE
alist 的奇思妙想

### DISCUSSION THREAD

**Discussion by oneisall8955** [AlistGo/alist#3046] [Ideas]
alist是大而全的，满足了大多数人的需求，很感激组织及各位Contributors的无私贡献。
使用alist不久，alist给我带来了很多便利，也让人脑洞大开，alist，或许能变得更强大？

以下想法可能会偏离alist的核心价值（多存储的文件列表），源自于我使用其他产品的经历。

### 功能插件化
参考utool建设插件市场，作为开放市场，同时提供官方插件，携带默认插件等
这是非常核心的价值，基于插件化，可实现的太多了。几个脑洞插件：

- 存储插件化，当前支持的类型非常多，可以按需安装进行支持
- 刮削插件，形如 TinyMediaManager
- 媒体影音插件，配合刮削，构建出在线的影音中心
- 文件管理&任务插件，组合一些任务，进行备份，迁移
- 主题插件，丰富 alist 的界面，形如typecho博客，功能和主题都插件化了
- web 静态服务解析及管理，在指定目录生成简单的博客，管理博文，处理临时文档等
- 资源供应商，如搜索网友共享，一键导入，一键挂载共享等
- 一起看，弹幕等视频播放插件（有点鸡肋）

显然，上述很多功能，将云盘挂载到本地，用本地专业的工具会更合适。但在插件交互做的足够好情况下，会帮助一些初级玩家节省时间，让操作停留在alist里面。插件不仅让alist架构更完整，同时也丰富了alist生态。

### 商业化
参考bitwarden，提供官方服务，免费版本及订阅版（vip开启高级功能），并且支持自建，自建的全开放。
让alist拥有saas能力，在官方注册账号，用户使用官方平台进行挂载云盘，免去用户投入硬件及软件配置时间
对于会员体系，可通过挂载云盘个数，创建的用户个数，同时在线使用session数目，安装的插件数量等限制

**Comment by anwen-anyi** [comment]
功能插件化先不说，商业化感觉会被起诉？QAQ

**Reply by NovaShen555** [reply]
+1

**Reply by lwb-2021** [reply]
我比较喜欢All in One，如果alist真能出插件的话我肯定先去搞个m3u8的离线下载（自己真的懒得做鉴权和文件共享，如果能用alist的组件库的话更好了）

**Comment by dianso** [comment]
我可以开发这些功能，估计的一年左右。

给我开基本工资就行 25K每月

有兴趣联系我
